bot/utils.py

```diff
@@ -184,6 +184,49 @@ async def decode_expense_params(params, group):
 
     return data
 
+async def decode_payments_params(params,user,group):
+    """
+    Process command params in payment's attributes, and return a dict with the following data:
+    amount = expense amount
+    date = date
+    to_user = User
+    """
+    result = {}
+    if len(params) == 3:
+        try:
+            date = parse_date(params.pop(-1), DATE_INPUT_FORMATS)
+        except DateFormatterError:
+            example_date = dt.date.today()
+            formated_dates = [dt.datetime.strftime(example_date, format) for format in DATE_INPUT_FORMATS]
+            date_bullets = '\n - '.join(formated_dates)
+            text = f'El formato de fecha no es correcto. ' \
+                   f'Por ejemplo a la fecha de hoy la podés escribir en cualquiera ' \
+                   f'de los siguientes formatos: \n - {date_bullets}'
+            raise ParameterError(text)
+    else:
+        date = dt.date.today().strftime("%Y-%m-%d")
+    result["date"] = date
+    try:
+        amount, to_user = params
+        amount = float(amount)
+        if to_user.startswith("@"):
+            to_user = to_user[1:]
+        to_user = await User.objects.exclude(pk=user.pk).aget(username=to_user, telegram_groups=group)
+        result["amount"] =amount
+        result["to_user"] = to_user
+    except ValueError:
+        text =  "El primer argumento debe ser el monto a pagar, y el segundo argumento el "\
+                "username del usuario al que le estás pagando. \n\n"\
+                "Opcionalmente puede contener un tercer argumento con la fecha en la que se "\
+                "desea  computar el gasto, con el formato dd/mm/yy."
+        return text
+    except User.DoesNotExist:
+        text = "El usuario espcificado ({}) no existe dentro de este grupo. \n".format(to_user)
+        text += "Los posibles usuarios a los que les podes cargar un pago son: \n"
+        async for member in group.users.exclude(pk=user.pk):
+            text += "- {}\n".format(member.username)
+        return text
```

Comment:
El resultado de la función debería ser un dict, entiendo que para estos dos casos, en lugar de hacer `return text` se debería raisear el error esperado:

```suggestion
                "desea  computar el gasto, con el formato dd/mm/yy."
        raise ParameterError(text)
    except User.DoesNotExist:
        text = "El usuario espcificado ({}) no existe dentro de este grupo. \n".format(to_user)
        text += "Los posibles usuarios a los que les podes cargar un pago son: \n"
        async for member in group.users.exclude(pk=user.pk):
            text += "- {}\n".format(member.username)
        raise ParameterError(text)
```

Reply:
Si, se ve que hice un cambio de ultima momento y quedo mal esa parte. Ahí se agrega el lanzamiento de excepciones como bien sugeriste. Gracias @sofide !